apps/extension/src/ui/domains/Sign/Substrate/xTokens/SubSignXTokensTransfer.tsx

```diff
@@ -9,13 +8,37 @@ import useChains from "@ui/hooks/useChains"
 import { useExtrinsic } from "@ui/hooks/useExtrinsic"
 import { useTokenRatesMap } from "@ui/hooks/useTokenRatesMap"
 import useTokens from "@ui/hooks/useTokens"
+import isEqual from "lodash/isEqual"
 import { useMemo } from "react"
 import { useTranslation } from "react-i18next"
+import * as $ from "subshape"
 
 import { SignContainer } from "../../SignContainer"
 import { usePolkadotSigningRequest } from "../../SignRequestContext"
+import { SignViewBodyShimmer } from "../../Views/SignViewBodyShimmer"
 import { SignViewIconHeader } from "../../Views/SignViewIconHeader"
 import { SignViewXTokensTransfer } from "../../Views/transfer/SignViewCrossChainTransfer"
+import { $versionedMultiLocation, VersionedMultiLocation } from "../shapes/VersionedMultiLocation"
+import { XcmV3Junction } from "../shapes/XcmV3Junction"
+
+const normalizeTokenId = (tokenId: unknown) => {
```

Comment:
Should we be able to know what the return type of this function is?

Reply:
Not easily, it can be an object that describes a token from another chain (xcm junction). 
The only caller is `isSameTokenId`` which compares 2 of them, so I don't think we should bother with typings here.
Doing it this way because in chain data this JSON information is stored as string with sometimes invalid casing.

Reply:
oh also, i didn't import types for token types with subshape as they differ on every chain.

Reply:
ex here : https://substrate-metadata-explorer.pages.dev/types?chain=polkadot-acala
type to select is acala_primitives::currency::CurrencyId

Reply:
ok sounds good!